### PR TITLE
[cd][simplify/2] deprecate cluster env

### PR DESCRIPTION
**Pull request by can-anyscale**
Release tests are fueled by byod 100%. Remove all logic for non-byod tests in the code.

Test:
- CI

### CODE REVIEW

File: release/ray_release/tests/test_cluster_manager.py [hunk 1]
```diff
@@ -100,26 +92,28 @@ def setUp(self) -> None:
...
                     "name": f"unit_test__{self.__class__.__name__}",
-                    "cluster": {},
+                    "cluster": {"byod": {}},
                 }
             ),
         )
         self.sdk.reset()
         self.sdk.returns["get_cloud"] = APIDict(result=APIDict(provider="AWS"))
+        os.environ["BUILDKITE_COMMIT"] = "123456"
+        os.environ["BUILDKITE_BRANCH"] = "master"
```
Comment (aslonnie) [inline release/ray_release/tests/test_cluster_manager.py:102]: why are these needed?

File: release/ray_release/tests/test_glue.py [hunk 2]
```diff
@@ -161,13 +165,13 @@ def mock_alerter(test: Test, result: Result):
...
-            cluster=dict(
-                cluster_env="cluster_env.yaml", cluster_compute="cluster_compute.yaml"
-            ),
+            cluster=dict(byod={}, cluster_compute="cluster_compute.yaml"),
             alert="unit_test_alerter",
         )
         self.anyscale_project = "prj_unit12345678"
         self.ray_wheels_url = "http://mock.wheels/"
+        os.environ["BUILDKITE_COMMIT"] = "123456"
+        os.environ["BUILDKITE_BRANCH"] = "master"
```
Comment (aslonnie) [inline release/ray_release/tests/test_glue.py:174]: same question as the other.

File: release/ray_release/tests/test_cluster_manager.py (hunk 1 above)
Comment (can-anyscale) [reply release/ray_release/tests/test_cluster_manager.py:102]: they are used for byod tests to name the byod docker image: https://github.com/ray-project/ray/blob/master/release/ray_release/test.py#L224; previously those tests didn't need to set these because it never execute that parts of the code ;)

Comment (aslonnie) [reply release/ray_release/tests/test_cluster_manager.py:102]: I worry that this changes the env permanently without changing back, and will make other tests harder.

could you:

- set up the env early, like even before `self.sdk = MockSDK()`
- saves the original env state of these keys.
- add a `tearDown()` function, and restore the env to its original state.

alternatively, you can also support some kind of env lookup function pointer, and send it into Test on construction, and load the env with that function. similar to what I did in rayci in go:

https://github.com/ray-project/rayci/blob/main/raycicmd/env.go#L15

**Comment by can-anyscale** [comment]
let me remove these from review queue and focus on civ2 migration first ;)

**Review by aslonnie — COMMENTED** [review]
(let me know when this is ready for review.)
